src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -743,6 +743,13 @@ protected function embed_links( $data, $embed = true ) {
 						continue;
 					}
 
+					if ( empty( $request['per_page'] ) ) {
+						$matched = $this->match_request_to_handler( $request );
+						if ( ! empty( $matched ) && ! is_wp_error( $matched ) && isset( $matched[1]['args']['per_page']['maximum'] ) ) {
```

Comment:
```suggestion
						if ( ! is_wp_error( $matched ) && isset( $matched[1]['args']['per_page']['maximum'] ) ) {
```

The empty check is superfluous.